libinteract/libinteract.py

```diff
@@ -866,156 +867,140 @@ def calc_cg_fullmatrix(identifiers, idxs, percmat, perco):
-    array = np.array(contact_list)
-    array_T = array.T
-    # Initialize output
-    array_list = [array]
-    # Find unique chains in the array
-    chains = np.unique(array_T[0])
+    table_rows = np.array(table)
+    table_cols = table_rows.T
+    # Initialize output dictionary of tables
+    table_dict = {"all": table_rows}
     # Choose which column has the second residue
-    if hb:
-        sec_res = 4
-    else:
-        sec_res = 3
+    sec_res = 4 if hb else 3
+    # Find unique chains in the table
+    chains = np.unique(np.concatenate((table_cols[0], table_cols[sec_res])))
     # If multiple chains are present, split the contacts by chain
     if len(chains) > 1:
 	# For each chain, add the nodes that are only in contact with the same chain
         for chain in chains:
-            filtered_array = filter_by_chain(chain, chain, sec_res, \
-                                            array, array_T)
-            if filtered_array is not None:
-                array_list.append(filtered_array)
+            filtered_rows = filter_by_chain(chain, chain, sec_res, table_rows)
+            if filtered_rows is not None:
+                table_dict[chain] = filtered_rows
         # Create vector of all nodes that are in contact with different chains
-        logical_vector = array_T[0] != array_T[sec_res]
-        if logical_vector.sum == 0:
-            warnstr = "No interchain contacts found" 
-            log.warning(warnstr)
+        logical_vector = table_cols[0] != table_cols[sec_res]
+        if logical_vector.sum() == 0:
+            log.warning("No interchain contacts found")
         else:
             # For all combinations of different chains, find nodes that are in
             # contact with different chains
             for chain1, chain2 in itertools.combinations(chains, 2):
                 # Check both combinations (e.g. A, B and B, A)
-                filtered_array1 = filter_by_chain(chain1, chain2, sec_res, \
-                                               array, array_T)
-                filtered_array2 = filter_by_chain(chain2, chain1, sec_res, \
-                                               array, array_T)
+                filtered_rows1 = filter_by_chain(chain1, chain2, sec_res, 
+                                                 table_rows)
+                filtered_rows2 = filter_by_chain(chain2, chain1, sec_res, 
+                                                 table_rows)
                 # If both outputs exist, concatenate and append
-                if filtered_array1 is not None and filtered_array2 is not None:
-                    filtered_array = np.concatenate((filtered_array1, \
-                                                     filtered_array2))
-                    array_list.append(filtered_array)
+                if filtered_rows1 is not None and filtered_rows2 is not None:
+                    filtered_rows = np.concatenate((filtered_rows1, 
```

Comment:
you should change the `filtered_rows` variable name in here - it's the same variable name you used earlier for something else. It's not a problem since you have stored the previous content of that variable (i.e. a reference to an array) in a dictionary that you then are returning, but it can be misleading when reading the code